quartz/components/scripts/explorer.inline.ts

```diff
@@ -121,6 +124,26 @@ function setupExplorer() {
     // If tree is not in localStorage or config is disabled, use tree passed from Explorer as dataset
     explorerState = JSON.parse(explorer.dataset.tree)
   }
+
+  // Check if localStorage fileTree structure needs to be cleared
+
+  if (!localStorage.getItem("fileTree") || !explorer?.dataset.tree) return
+  // Get non localStorage explorer state and pick out paths
+  const realState: FolderState[] = JSON.parse(explorer.dataset.tree)
+  const explorerPaths = realState.map((entry) => entry.path)
+
+  // Create hash from stringified explorer state
+  const hash = await hashString(JSON.stringify(explorerPaths))
+
+  const storageHash = localStorage.getItem(localStorageHashName)
+
+  // Update localStorage if it needs updating (hash + fileTree state)
+  if (!storageHash) {
+    localStorage.setItem(localStorageHashName, hash)
+  } else if (storageHash !== hash) {
+    localStorage.setItem(localStorageHashName, hash)
+    localStorage.setItem("fileTree", JSON.stringify(realState))
+  }
```

Comment:
```suggestion
  if (storageHash !== hash) {
    localStorage.setItem(localStorageHashName, hash)
    localStorage.setItem("fileTree", JSON.stringify(realState))
  }
```

strictly more general (i think `!storageHash` would also result in `fileTree` not being stored)